src/api/procedures/consumeAuthorizationRequests.ts

```diff
@@ -119,6 +125,8 @@ export async function getAuthorization(
     if (!accept) {
       did = await fetchDid();
       condition = condition || did === issuer.did;
+    } else {
+      await assertAuthorizationRequestValid(context, authRequest);
```

Comment:
These sorts of validations should go in the prepare method. The `getAuthorization` method is only for permissions and roles